src/services/PwmService/PwmService.ts

```diff
@@ -0,0 +1,191 @@
+import * as protobuf from "protobufjs";
+import path from "path";
+import { RpcChannel } from "../../rpcChannel/RpcChannel";
+import type {
+  serverResponse,
+  serviceRequest,
+} from "../../rpcChannel/ReqRepTypes";
+import { PWMConfig, PWM, Polarity } from "../PwmService/PwmTypes";
+import { createConfigArgsList } from "../util/helpers";
+import { SuccessMsg, GetDutyCycle, GetFrequency, GetPolarity, GetEnabled } from '../rpcServiceTypes';
+import { Console } from "console";
+
+const protoPckgPath = path.join(require.resolve('@edgepi-cloud/edgepi-rpc-protobuf'), '../edgepi_rpc_protos');
+
+const SUCCESS_MSG = "EdgePiRPC_PWM.SuccessMsg";
+const GET_FREQUENCY = "EdgePiRPC_PWM.GetFrequency";
+const GET_DUTY_CYCLE = "EdgePiRPC_PWM.GetDutyCycle";
+const GET_POLARITY = "EdgePiRPC_PWM.GetPolarity";
+const GET_ENABLED = "EdgePiRPC_PWM.GetEnabled";
+const PWM_TYPE = "EdgePiRPC_PWM.PWM";
+const CONFIG_TYPE = "EdgePiRPC_PWM.Config";
+
+/**
+ * @constructor PWM class for calling EdgePi PWM SDK methods through RPC
+ */
+class PWMService {
+  rpcProtoRoot: protobuf.Root;
+  serviceProtoRoot: protobuf.Root;
+  serviceName: string;
+  rpcChannel: RpcChannel;
+
+  constructor(serverEndpoint: string) {
+    this.rpcProtoRoot = protobuf.loadSync(
+      path.join(protoPckgPath, "rpc.proto")
+    );
+    this.serviceProtoRoot = protobuf.loadSync(
+      path.join(protoPckgPath, "pwm.proto")
+    );
+    this.serviceName = "PWMService";
+    this.rpcChannel = new RpcChannel(serverEndpoint, this.rpcProtoRoot);
+    console.info(this.serviceName, "initialized");
+  }
+
+  /**
+ * @async Calls the PWM init_pwm SDK method through an RPC request
+ */
+  async initPwm(PWM: PWM): Promise<string> {
+    const response = await this.PWMRequest("init_pwm", PWM, SUCCESS_MSG);
+    return (response.content as SuccessMsg).content;
+  }
+
+  /**
+   * @async Calls the PWM enable SDK method through an RPC request
+   */
+  async enable(PWM: PWM): Promise<string> {
+    const response = await this.PWMRequest("enable", PWM, SUCCESS_MSG);
+    return (response.content as SuccessMsg).content;
+  }
+
+  /**
+   * @async Calls the PWM disable SDK method through an RPC request
+   */
+  async disable(PWM: PWM): Promise<string> {
+    const response = await this.PWMRequest("disable", PWM, SUCCESS_MSG);
+    return (response.content as SuccessMsg).content;
+  }
+
+  /**
+   * @async Calls the PWM close SDK method through an RPC request
+   */
+  async close(PWM: PWM): Promise<string> {
+    const response = await this.PWMRequest("close", PWM, SUCCESS_MSG);
+    return (response.content as SuccessMsg).content;
+  }
+
```

Comment:
Will all the methods ( enable, disable, close, ...) be asynchronously executed in this function? Are we supposed to see `SuccessMsg` logs in the console?

Reply:
All methods are asynchronously executed. `SuccessMsg` is not logged in the console.